tests/product.spec.ts

```diff
@@ -32,4 +34,100 @@ test.describe('Products', () => {
             'Add to favorites button is not visible',
         ).toBeVisible();
     });
+
+    test('Verify user can add product to cart', async ({ page }) => {
+        const app = new Application(page);
+        const productName: string = 'Slip Joint Pliers';
+        const textAllert: string = 'Product added to shopping cart.';
+
+        await app.home.navigateTo('/');
+        const productDetails: {title: string, price: string} = await app.home.getProductDetails(productName);
+        await app.home.openProduct(productName);
+        await expect(page).toHaveURL(/^https:\/\/practicesoftwaretesting.com\/product/);
+        await expect(app.productDetails.productName, 'Product name is not visible').toHaveText(productDetails.title);
+        await expect(app.productDetails.productPrice, 'Product price is not visible').toHaveText(productDetails.price);
+
+        await app.productDetails.addToCartButton.click();
+        await expect(app.productDetails.productAddedAlertMessage).toBeVisible();
+
+        const alertMessage: string = await app.productDetails.getTextElement(app.productDetails.productAddedAlertMessage);
+        expect(alertMessage).toEqual(textAllert);
```

Comment:
```suggestion
        await expect(app.productDetails.productAddedAlertMessage).toHaveText(textAllert, { useInnerText: true });
```
That is what I mentioned above. Using asynchronous assertion whenever possible will make your test much more stable, especially for validating text on the page.

Do the same for other similar assertions in the test